### DISCUSSION TITLE
shell文件格式化插件`vs-shell-format`,格式化文档功能失效

### DISCUSSION THREAD

**Discussion by yesszh** [opensumi/core#1717] [Q&A] [answered]
1. 在sumi IDE中安装`vs-shell-format`插件，右击脚本选择【格式化文档】 功能没有用，但是选择【使用...格式化文档】
2. 在vscode中可以正常使用`vs-shell-format`插件【格式化文档】 功能，请问下是不是sumi不兼容这个插件还是其他问题导致的？

**Comment by yesszh** [comment]
补充下：对bash脚本 右击脚本选择【格式化文档】 功能没有用，但是选择【使用...格式化文档】却可以格式化有用

**Reply by erha19** [reply]
@yesszh 这两个菜单有行为上的区别，`格式化文档` 菜单，会使用你设置内保存的值作为格式化选项，而 `使用...格式化文档` 采用的则是弹窗让用户选择的方式。你这边使用的是 https://marketplace.visualstudio.com/items?itemName=foxundermoon.shell-format 这个插件吗？

**Reply by yesszh** [reply]
是上面那个插件，使用这个插件点击格式化文档是没有反应的,也没提示让设置内保存的值

**Reply by erha19** [reply]
@yesszh 方便提供一下你这边测试使用的脚本吗？我这边跟进看下

**Reply by yesszh** [reply]
安装了其他格式化插件【SQL Beautify】， 这个右击选择 ‘格式化文档’ 菜单，这个功能又正常格式化使用
https://marketplace.visualstudio.com/items?itemName=clarkyu.vscode-sql-beautify

**Reply by yesszh** [reply]
> @yesszh 方便提供一下你这边测试使用的脚本吗？我这边跟进看下
我是直接用的是bash 注释多行下，然后缩进不对齐，再使用格式化插件格式

**Reply by erha19** [reply]
看了一下，这个插件是存在兼容性问题的，执行格式化过程时会有一个 `Error: write EPIPE` 的插件进程报错，我明天排查一下 @yesszh  你可以看一下是否有其他替代插件可以使用

**Reply by yesszh** [reply]
@erha19 好的，这个比较常用的插件希望可以兼容处理

**Comment by erha19** [comment, marked as answer]
@yesszh 我这边重新测试了一下，发现再 2.20.0+ 版本下是可以用的，你那边使用的是哪个版本？

https://user-images.githubusercontent.com/9823838/192676713-cb28b85d-2b0a-49c1-aa13-0dda84d46978.mp4

**Reply by yesszh** [reply]
@erha19 
系统 macos:12.15.1
浏览器 chrome:105.0.5195.12
opensumi:2.19.10 项目中应用的版本
shell-format: 7.2.2 

单独试了下ide-startup,同时删除之前初始化的.sumi目录，再应用市场中安装插件；
选择【格式化文档】还是没有反应，也没有看到前后端打印报错信息；
![image](https://user-images.githubusercontent.com/35323838/192680905-9d7f8df0-3b1d-4d70-81b4-44c5b5472825.png)
![image](https://user-images.githubusercontent.com/35323838/192681274-8224b500-e836-483b-9776-4a6a51f136ee.png)

**Reply by erha19** [reply]
@yesszh 你这边在 ide-startup pull 一下最新 master 分支，看起来该问题在 2.20.0 版本后才被修复了

**Reply by yesszh** [reply]
@erha19 👍🏿  ide-startup更新到最新版本是可以啦，后面我项目也更新看下

**Reply by erha19** [reply]
@yesszh 升级参考文档 https://github.com/opensumi/core/wiki/%E9%9B%86%E6%88%90%E6%8E%A5%E5%85%A5-Breaking-Changes#v2200
